src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -119,58 +85,62 @@ protected function mapPluginsByResourceName(): void
+                )
+            );
         }
 
-        return $effectivePlugins;
+        return $filteredPlugins;
     }
 
     /**
-     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $effectivePlugins
-     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $pluginsPerExporter
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
      *
-     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
+     * @return bool
      */
-    protected function extractEffectivePlugins($effectivePlugins, $pluginsPerExporter): array
+    protected function isRepositoryPlugin(EventResourcePluginInterface $plugin): bool
     {
-        foreach ($effectivePlugins as $effectivePlugin) {
-            if ($effectivePlugin instanceof EventResourceRepositoryPluginInterface || $effectivePlugin instanceof EventResourceBulkRepositoryPluginInterface) {
-                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
-            }
-
-            if ($effectivePlugin instanceof EventResourceQueryContainerPluginInterface) {
-                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
-            }
-        }
-
-        $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS]);
-        $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS]);
+        return ($plugin instanceof EventResourceRepositoryPluginInterface || $plugin instanceof EventResourceBulkRepositoryPluginInterface);
+    }
 
-        return $pluginsPerExporter;
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
+     *
+     * @return bool
+     */
+    protected function isQueryContainerPlugin(EventResourcePluginInterface $plugin): bool
+    {
+        return ($plugin instanceof EventResourceQueryContainerPluginInterface);
     }
 
     /**
-     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $eventResourcePlugins
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
      *
-     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
+     * @return bool
      */
-    protected function resolveDuplicatePlugins(array $eventResourcePlugins): array
+    protected function uniqueResourceAndEventNamesCallback(EventResourcePluginInterface $plugin): bool
     {
-        return array_map('current', $eventResourcePlugins);
+        static $eventList = [];
```

Comment:
Please move it to class level.

Reply:
For what?

Reply:
Static variables inside the function for instances of the same class remain their state.
See https://stackoverflow.com/questions/6188994/static-keyword-inside-function

Reply:
Hmm, do you consider the case when you function will be called twice? 
it doesn't seem that it will work properly